src/sql/parts/notebook/models/notebookModel.ts

```diff
@@ -380,45 +395,48 @@ export class NotebookModel extends Disposable implements INotebookModel {
 	}
 
 	public doChangeKernel(kernelSpec: nb.IKernelSpec): Promise<void> {
-		this.findProviderIdForKernel(kernelSpec);
-		return this._activeClientSession.changeKernel(kernelSpec)
-			.then((kernel) => {
-				kernel.ready.then(() => {
-					if (kernel.info) {
-						this.updateLanguageInfo(kernel.info.language_info);
-					}
-				}, err => undefined);
-				return this.updateKernelInfo(kernel);
-			}).catch((err) => {
-				this.notifyError(localize('changeKernelFailed', 'Failed to change kernel: {0}', notebookUtils.getErrorMessage(err)));
-				// TODO should revert kernels dropdown
-			});
+		this.setProviderIdForKernel(kernelSpec);
+		if (this._activeClientSession && this._activeClientSession.isReady) {
+			return this._activeClientSession.changeKernel(kernelSpec)
+				.then((kernel) => {
+					this.updateKernelInfo(kernel);
+					kernel.ready.then(() => {
+						if (kernel.info) {
+							this.updateLanguageInfo(kernel.info.language_info);
+						}
+					}, err => undefined);
+				}).catch((err) => {
+					this.notifyError(localize('changeKernelFailed', 'Failed to change kernel: {0}', notebookUtils.getErrorMessage(err)));
+					// TODO should revert kernels dropdown
+				});
+		}
+		return Promise.resolve();
```

Comment:
What's this case? Is it initiating a new session, and will the kernel ever be set?

Reply:
The case is "no active client session exists". I can add an error message if you think it's helpful.